lib/finch.ex

```diff
@@ -147,10 +149,24 @@ defmodule Finch do
   end
 
   defp valid_opts_to_map(valid) do
+    # We need to enable keepalive and set the nodelay flag to true by default.
+    transport_opts =
+      valid
+      |> get_in([:conn_opts, :transport_opts])
+      |> List.wrap()
+      |> Keyword.put_new(:timeout, @default_connect_timeout)
+      |> Keyword.put_new(:nodelay, true)
```

Comment:
This might be a bad default to set for bandwidth for users that send many small requests in sequence through HTTP/1 pipelining or on HTTP/2, but can be good to set if you want low latency for individual requests. Why did this help with the connection closed error?

Reply:
It didn't. I was just already in here adding stuff. I think this is the correct default value for the majority of cases. I'll add documentation that explains that people may want to disable this if they're worried about network congestion.

Reply:
Also worth mentioning, Hackney sets `keepalive: true` and `nodelay: true` by default too: https://github.com/benoitc/hackney/blob/039db6033b9f81d75c9b0b672f9f8b96e9111575/src/hackney_http_connect.erl#L62